src/game/board.py

```
@@ -139,8 +158,9 @@ def set_cell(self, x: int, y: int, color: Optional[Tuple[int, int, int]]):
             y: Row index
             color: RGB color tuple or None for empty
         """
-        if 0 <= x < self.width and 0 <= y < self.height:
-            self.grid[y][x] = color
+        if not (0 <= x < self.width and 0 <= y < self.height):
+            raise IndexError(f"set_cell out of bounds: ({x}, {y})")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)